packages/api/tests/models/test-rules-model.js

```diff
@@ -149,10 +161,27 @@ test('Creating a rule with an invalid type throws an error', async (t) => {
 
   await t.throwsAsync(
     () => rulesModel.create(ruleItem),
-    { message: 'Rule type \'invalid\' not supported.' }
+    { name: 'SchemaValidationError' }
   );
 });
 
+test.serial('Creating an invalid rule does not create workflow triggers', async (t) => {
+  const { onetimeRule } = t.context;
+  const ruleItem = cloneDeep(onetimeRule);
```

Comment:
Nit: You don't need to cloneDeep

Reply:
I'm curious about this. Why is `cloneDeep` not necessary?

I see that ava does attempt to isolate data between tests (https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#test-context).

However, it's not clear that a deep clone is performed on the context. Specifically:

> Context created in .before() hooks is cloned before it is passed to .beforeEach() hooks and / or tests. The .after() and .after.always() hooks receive the original context value.
>
> For .beforeEach(), .afterEach() and .afterEach.always() hooks the context is not shared between different tests, allowing you to set up data such that it will not leak to other tests.

The first sentence links to lodash's shallow clone (see the ava docs at the link above), but the second sentence seems to _imply_ that some sort of deep clone is happening, but it is not clear as to what actually happens.

Reply:
`t.context.onetimeRule` is created in `beforeEach`. And as the docs you posted say, context set up in beforeEach is not shared between tests. So there's no need to clone deep because `t.context.onetimeRule` will already be a new object in each test

Reply:
Oh, I see. That makes sense now. Thanks!